numpy/_core/src/umath/string_buffer.h

```diff
@@ -1505,4 +1505,81 @@ string_expandtabs(Buffer<enc> buf, npy_int64 tabsize, Buffer<enc> out)
 }
 
 
+enum class JUSTPOSITION {
+    CENTER, LEFT, RIGHT
+};
+
+template <ENCODING enc>
+static inline npy_intp
+string_pad(Buffer<enc> buf, npy_int64 width, Buffer<enc> fill, JUSTPOSITION pos, Buffer<enc> out)
+{
+    size_t finalwidth = width > 0 ? width : 0;
+    size_t len = buf.num_codepoints();
+    if (len >= finalwidth) {
+        buf.buffer_memcpy(out, len);
+        out.buffer_fill_with_zeros_after_index(len);
+        return (npy_intp) len;
+    }
+
+    size_t left, right;
+    if (pos == JUSTPOSITION::CENTER) {
+        size_t pad = finalwidth - len;
+        left = pad / 2 + (pad & finalwidth & 1);
+        right = pad - left;
+    }
+    else if (pos == JUSTPOSITION::LEFT) {
+        left = 0;
+        right = finalwidth - len;
+    }
+    else {
+        left = finalwidth - len;
+        right = 0;
+    }
+
+    if (left == 0 && right == 0) {
+        buf.buffer_memcpy(out, len);
+        out.buffer_fill_with_zeros_after_index(len);
+        return len;
+    }
+
+    if (left > PY_SSIZE_T_MAX - len || right > PY_SSIZE_T_MAX - (left + len)) {
```

Comment:
This should also not be possible, at least if we check `if (finalwidth > PY_SSIZE_T_MAX)` on top.

Reply:
Would you prefer having a check like that on top instead of this? Cause we currently don't.

Reply:
Yes, that seemed to make sense, to just get it out of the way.